src/core/internal/array/comparison.d

```diff
@@ -60,26 +60,24 @@ int __cmp(T)(scope const T[] lhs, scope const T[] rhs) @trusted
         immutable len = lhs.length <= rhs.length ? lhs.length : rhs.length;
         foreach (const u; 0 .. len)
         {
-            static if (__traits(isFloating, T))
-            {
-                immutable a = lhs.ptr[u], b = rhs.ptr[u];
-                static if (is(T == cfloat) || is(T == cdouble)
+            auto a = lhs.ptr[u], b = rhs.ptr[u];
+            static if (__traits(isFloating, T) && is(T == cfloat) || is(T == cdouble)
```

Comment:
I suggest replacing the entire condition with the more succinct
```d
static if (is(T : creal))
````
 I have verified that for scalar `T` this has no performance penalty relative to
```d
static if (__traits(isFloating, T) && (is(T == cfloat) || is(T == cdouble) || is(T == creal)))
```
(in fact it is ever-so-slightly faster and ever-so-slightly more memory efficient).

Reply:
OK